### PR TITLE
[docs] Add docs for DeepSpeed Infinity

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

This PR adds docs for the NVMe Offloading in the DeepSpeed plugin

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [x] Did you list all the breaking changes introduced by this pull request?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: docs/source/advanced/advanced_gpu.rst [hunk 1]
```diff
@@ -423,11 +423,49 @@ DeepSpeed ZeRO Stage 3 Offloads optimizer state, gradients to the host CPU to re
...
+            remote_device='nvme',
+            offload_params_device='nvme',
+            offload_optimizer_device='nvme',
+            nvme_path = '/local_nvme'
+        ),
         precision=16
     )
     trainer.fit(model)
 
+When offloading to NVMe you may notice that the speed is slow. There are parameters that need to be tuned based on the drives that you are using. Running the `aio_bench_perf_sweep.py <https://github.com/microsoft/DeepSpeed/blob/master/csrc/aio/py_test/aio_bench_perf_sweep.py>`__ script can help you to find optimum parameters. See the `issue <https://github.com/microsoft/DeepSpeed/issues/998>`__ for more information on how to parse the information.
```
Comment (awaelchli) [inline docs/source/advanced/advanced_gpu.rst:468]:
```suggestion
When offloading to NVMe you may notice that the speed is slow. There are parameters that need to be tuned based on the drives that you are using. Running the `aio_bench_perf_sweep.py <https://github.com/microsoft/DeepSpeed/blob/master/csrc/aio/py_test/aio_bench_perf_sweep.py>`__ script can help you to find optimal parameters. See the `issue <https://github.com/microsoft/DeepSpeed/issues/998>`__ for more information on how to parse the information.
```
optimal?

**Review by awaelchli — APPROVED** [review]
nice

File: docs/source/advanced/advanced_gpu.rst [hunk 2]
```diff
@@ -423,11 +423,49 @@ DeepSpeed ZeRO Stage 3 Offloads optimizer state, gradients to the host CPU to re
...
+    trainer = Trainer(
+        gpus=4,
+        plugins=DeepSpeedPlugin(
+            stage=3,
+            offload_optimizer=True,
+            offload_parameters=True,
+            remote_device='nvme',
+            offload_params_device='nvme',
+            offload_optimizer_device='nvme',
+            nvme_path = '/local_nvme'
```
Comment (kaushikb11) [inline docs/source/advanced/advanced_gpu.rst:462]: 🔥

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/advanced/advanced_gpu.rst (hunk 1 above)
Comment (kaushikb11) [reply docs/source/advanced/advanced_gpu.rst:468]: Apparently, optimum is also an adjective.

Comment (awaelchli) [reply docs/source/advanced/advanced_gpu.rst:468]: I'll take it back then :) 
I'm not native speaker <3
